src/apdu_handler.c

```diff
@@ -0,0 +1,359 @@
+    if (showAddr) {
+        view_review_init(addr_getItem, addr_getNumItems, app_reply_ok);
+        view_review_show(REVIEW_ADDRESS);
+        *flags |= IO_ASYNCH_REPLY;
+        return;
+    }
+    
+    *tx = action_addrResponseLen;
+
+    THROW(APDU_CODE_OK);
+
+    return;
+}
+
+
+__Z_INLINE void handleSignSecp256K1(volatile uint32_t *flags, volatile uint32_t *tx, uint32_t rx){
+    if (!process_chunk(rx, true))
+        THROW(APDU_CODE_OK);
+
+    const char *error_msg = tx_parse();
+
+    if (error_msg != NULL) {
+        int error_msg_length = strlen(error_msg);
+        MEMCPY(G_io_apdu_buffer, error_msg, error_msg_length);
+        *tx += (error_msg_length);
+        THROW(APDU_CODE_DATA_INVALID);
+    }
+
+    CHECK_APP_CANARY()
+    view_review_init(tx_getItem, tx_getNumItems, tx_accept_sign);
+    view_review_show(REVIEW_TXN);
+    *flags |= IO_ASYNCH_REPLY;
+}
+
+void handleApdu(volatile uint32_t *flags, volatile uint32_t *tx, uint32_t rx) {
+    uint16_t sw = 0;
+
+    BEGIN_TRY
+    {
+        TRY
+        {
+            if (G_io_apdu_buffer[OFFSET_CLA] != CLA) {
+                THROW(APDU_CODE_CLA_NOT_SUPPORTED);
+            }
+
+            if (rx < 5) {
+                THROW(APDU_CODE_WRONG_LENGTH);
+            }
+            switch (G_io_apdu_buffer[OFFSET_INS]) {
+                case INS_GET_VERSION: {
+                    handleGetVersion(tx);
+                    break;
+                }
+
+                // INS_PUBLIC_KEY_SECP256K1 will be deprecated in the near future
+                case INS_PUBLIC_KEY_SECP256K1: {
+                    handleGetUncompressedPubKey(tx, rx);
+                    break;
+                }
+
+                case INS_SHOW_ADDR_SECP256K1: {
+                    handleGetAddrSecp256K1(flags, tx, rx, true);
+                    break;
+                }
+
+                case INS_GET_ADDR_SECP256K1: {
+                    handleGetAddrSecp256K1(flags, tx, rx, false);
+                    break;
+                }
+
+                case INS_SIGN_SECP256K1: {
+                    handleSignSecp256K1(flags, tx, rx);
+                    break;
+                }
+
+#ifdef TESTING_ENABLED
+                case INS_HASH_TEST: {
+                    if (process_chunk(rx, false)) {
+                        uint8_t message_digest[CX_SHA256_SIZE];
+
+                        cx_hash_sha256(tx_get_buffer(),
+                                       tx_get_buffer_length(),
+                                       message_digest,
+                                       CX_SHA256_SIZE);
+
+                        memmove(G_io_apdu_buffer, message_digest, CX_SHA256_SIZE);
+                        *tx += 32;
+                    }
+                    THROW(APDU_CODE_OK);
+                }
+                break;
+                case INS_PUBLIC_KEY_SECP256K1_TEST: {
+                    // Generate key
```

Comment:
What is different here from function `handleGetUncompressedPubKey` above?

Reply:
Difference is we don't extract a derivation path from the command APDU... We use the default one that is set when the app is started.

Reply:
I don't think these test commands have much value now that we have functional tests. Keeping them for legacy...